include/Astro/orbitalElementConversions.hpp

```diff
@@ -105,14 +106,14 @@ Vector6 convertCartesianToKeplerianElements(
 
     // Set position and velocity vectors.
     std::vector< Real > position( 3 );
-    position[ 0 ] = cartesianElements[ 0 ];
-    position[ 1 ] = cartesianElements[ 1 ];
-    position[ 2 ] = cartesianElements[ 2 ];
+    position[ xPositionIndex ] = cartesianElements[ xPositionIndex ];
+    position[ yPositionIndex ] = cartesianElements[ yPositionIndex ];
+    position[ zPositionIndex ] = cartesianElements[ zPositionIndex ];
 
     std::vector< Real > velocity( 3 );
-    velocity[ 0 ] = cartesianElements[ 3 ];
-    velocity[ 1 ] = cartesianElements[ 4 ];
-    velocity[ 2 ] = cartesianElements[ 5 ];
+    velocity[ 0 ] = cartesianElements[ xVelocityIndex ];
```

Comment:
velocity[ 0 ] = cartesianElements[ xVelocityIndex ]; should be velocity[ xVelocityIndex ] = cartesianElements[ xVelocityIndex ]; 
can you also do the same change for the other two components as well